docs/sources/flow/reference/components/prometheus.operator.podmonitors.md

```diff
@@ -0,0 +1,193 @@
+configuration with the service account of the running Grafana Agent pod is
+used.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`api_server` | `string` | URL of the Kubernetes API server. | | no
+`kubeconfig_file` | `string` | Path of the `kubeconfig` file to use for connecting to Kubernetes. | | no
+`bearer_token_file` | `string` | File containing a bearer token to authenticate with. | | no
+`proxy_url` | `string` | HTTP proxy to proxy requests through. | | no
+`follow_redirects` | `bool` | Whether redirects returned by the server should be followed. | `true` | no
+`enable_http2` | `bool` | Whether HTTP2 is supported for requests. | `true` | no
+
+ At most one of the following can be provided:
+ - [`bearer_token` argument][client].
+ - [`bearer_token_file` argument][client].
+ - [`basic_auth` block][basic_auth].
+ - [`authorization` block][authorization].
+ - [`oauth2` block][oauth2].
+
+### basic_auth block
+
+{{< docs/shared lookup="flow/reference/components/basic-auth-block.md" source="agent" >}}
+
+### authorization block
+
+{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" >}}
+
+### oauth2 block
+
+{{< docs/shared lookup="flow/reference/components/oauth2-block.md" source="agent" >}}
+
+### tls_config block
+
+{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}
+
+### selector block
+
+The `selector` block describes a Kubernetes label selector for `PodMonitors`.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`match_labels` | `map(string)` | Label keys and values used to discover resources. | `{}` | no
+
+When the `match_labels` argument is empty, all resources will be matched.
+
+### match_expression block
+
+The `match_expression` block describes a Kubernetes label match expression for
+`PodMonitors` discovery.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`key` | `string` | The label name to match against. | | yes
+`operator` | `string` | The operator to use when matching. | | yes
+`values`| `list(string)` | The values used when matching. | | no
+
+The `operator` argument must be one of the following strings:
+
+* `"In"`
+* `"NotIn"`
+* `"Exists"`
+* `"DoesNotExist"`
+
+`selector` can make use of multiple
+`match_expression` inner blocks which are treated as AND clauses.
+
+## Exported fields
+
+`prometheus.operator.podmonitors` does not export any fields. It will forward all metrics it scrapes to the receiver configures with the `forward_to` argument.
+
+## Component health
+
+`prometheus.operator.podmonitors` is reported as unhealthy if any of its critical components fails during initialization (discovery manager, informers).
```

Comment:
I'd like for this to not reference as much internal names, since those can change easily:

```suggestion
`prometheus.operator.podmonitors` is reported as unhealthy when given an invalid configuration, Prometheus components fail to initialize, or the connection to the Kubernetes API could not be established properly.
```